ga4gh/datamodel/biodata.py

```diff
@@ -0,0 +1,114 @@
+"""
+Dataset objects
+"""
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import datetime
+
+import ga4gh.datamodel as datamodel
+import ga4gh.protocol as protocol
+import ga4gh.exceptions as exceptions
+
+
+class AbstractBioSample(datamodel.DatamodelObject):
+    """
+    This class represents an abstract BioSample object.
+    It sets default values and getters, as well as the
+    toProtocolElement function.
+    """
+    compoundIdClass = datamodel.BioSampleCompoundId
+
+    def __init__(self, parentContainer, localId):
+        super(AbstractBioSample, self).__init__(parentContainer, localId)
+        self._createDateTime = datetime.datetime.now().isoformat()
```

Comment:
If there is a createDateTime or updateDateTime in the JSON it will use that one. https://github.com/ga4gh/server/pull/967/files#diff-aaf9d6c43122e379253eff9dfda7d07cR85